modules/test_infrastructure/main.tf

```diff
@@ -0,0 +1,214 @@
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group
+resource "azurerm_resource_group" "this" {
+  count    = var.create_resource_group ? 1 : 0
+  name     = var.resource_group_name
+  location = var.region
+
+  tags = var.tags
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/resource_group
+data "azurerm_resource_group" "this" {
+  count = var.create_resource_group ? 0 : 1
+  name  = var.resource_group_name
+}
+
+locals {
+  resource_group = var.create_resource_group ? azurerm_resource_group.this[0] : data.azurerm_resource_group.this[0]
+}
+
+# https://registry.terraform.io/modules/PaloAltoNetworks/swfw-modules/azurerm/latest/submodules/vnet
+module "vnet" {
+  source = "../vnet"
+
+  for_each = var.vnets
+
+  name                   = each.value.name
+  create_virtual_network = each.value.create_virtual_network
+  resource_group_name    = local.resource_group.name
+  region                 = var.region
+
+  address_space = each.value.address_space
+
+  create_subnets = each.value.create_subnets
+  subnets        = each.value.subnets
+
+  network_security_groups = each.value.network_security_groups
+  route_tables            = each.value.route_tables
+
+  tags = var.tags
+}
+
+# https://registry.terraform.io/modules/PaloAltoNetworks/swfw-modules/azurerm/latest/submodules/vnet_peering
+module "vnet_peering" {
+  source   = "../vnet_peering"
+  for_each = { for k, v in var.vnets : k => v if v.hub_vnet_name != null }
```

Comment:
Should we have new line between `source` and `for_each`?

Reply:
We do it in the examples so I think we can stick to that.